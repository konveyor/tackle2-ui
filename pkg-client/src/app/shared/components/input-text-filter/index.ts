export { InputTextFilter } from "./input-text-filter";
