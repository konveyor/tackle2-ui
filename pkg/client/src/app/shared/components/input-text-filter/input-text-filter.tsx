import React, { useState } from "react";
import { useTranslation } from "react-i18next";

import { Button, InputGroup, TextInput } from "@patternfly/react-core";
import { SearchIcon } from "@patternfly/react-icons/dist/esm/icons/search-icon";

export interface InputTextFilterProps {
  onApplyFilter: (filterText: string) => void;
}

export const InputTextFilter: React.FC<InputTextFilterProps> = ({
  onApplyFilter,
}) => {
  const { t } = useTranslation();

  const [filterText, setFilterText] = useState("");

  const handleOnChangeFilterText = (value: string) => {
    setFilterText(value);
  };

  const handleOnSearch = () => {
    if (filterText.trim().length > 0) {
      onApplyFilter(filterText.trim());
      setFilterText("");
    }
  };

  const handleOnSearchKeyPress = (event: React.KeyboardEvent) => {
    if (event.key === "Enter") {
      handleOnSearch();
    }
  };

  return (
    <InputGroup>
      <TextInput
        type="text"
        value={filterText}
        onChange={handleOnChangeFilterText}
        onKeyPress={handleOnSearchKeyPress}
        placeholder={t("terms.filter")}
        aria-label="filter-text"
      />
      <Button variant="control" aria-label="search" onClick={handleOnSearch}>
        <SearchIcon />
      </Button>
    </InputGroup>
  );
};
