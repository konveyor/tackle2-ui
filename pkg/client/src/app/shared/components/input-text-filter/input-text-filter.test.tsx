import React from "react";
import { shallow, mount } from "enzyme";
import { InputTextFilter } from "./input-text-filter";

describe("InputTextFilter", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<InputTextFilter onApplyFilter={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip("Call 'onApplyFilter' using search btn", () => {
    const onApplyFilterSpy = jest.fn();
    const wrapper = mount(<InputTextFilter onApplyFilter={onApplyFilterSpy} />);

    // Change search input text
    const searchInput = wrapper.find("input");
    (searchInput.getDOMNode() as HTMLInputElement).value = "my filter text";
    searchInput.simulate("change");

    // Apply filter
    wrapper.find("button[aria-label='search']").simulate("click");

    // Verify
    expect(onApplyFilterSpy).toHaveBeenCalledWith("my filter text");
  });

  it.skip("Call 'onApplyFilter' using {enter}", () => {
    const onApplyFilterSpy = jest.fn();
    const wrapper = mount(<InputTextFilter onApplyFilter={onApplyFilterSpy} />);

    // Change search input text
    const searchInput = wrapper.find("input");
    (searchInput.getDOMNode() as HTMLInputElement).value = "my filter text";
    searchInput.simulate("change");

    // Apply filter
    wrapper.find("input").simulate("keypress", { key: "Enter" });

    // Verify
    expect(onApplyFilterSpy).toHaveBeenCalledWith("my filter text");
  });
});
