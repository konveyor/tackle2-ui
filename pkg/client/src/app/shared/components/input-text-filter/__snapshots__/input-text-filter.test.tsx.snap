// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextFilter Renders without crashing 1`] = `
<InputGroup>
  <TextInput
    aria-label="filter-text"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="terms.filter"
    type="text"
    value=""
  />
  <Button
    aria-label="search"
    onClick={[Function]}
    variant="control"
  >
    <Component />
  </Button>
</InputGroup>
`;
